packages/rocketchat-integrations/server/processWebhookMessage.js

```diff
@@ -22,41 +64,26 @@ this.processWebhookMessage = function(messageObj, user, defaultValues) {
 
 		switch (channelType) {
 			case '#':
-				room = RocketChat.models.Rooms.findOneByIdOrName(channel);
-				if (!_.isObject(room)) {
-					throw new Meteor.Error('invalid-channel');
-				}
-				rid = room._id;
-				if (room.t === 'c') {
-					Meteor.runAsUser(user._id, function() {
-						return Meteor.call('joinRoom', room._id);
-					});
-				}
+				room = retrieveRoomInfo({ userId: user._id, channel });
 				break;
 			case '@':
-				roomUser = RocketChat.models.Users.findOne({
-					$or: [
-						{
-							_id: channel
-						}, {
-							username: channel
-						}
-					]
-				}) || {};
-				rid = [user._id, roomUser._id].sort().join('');
-				room = RocketChat.models.Rooms.findOneById({$in: [rid, channel]});
-				if (!_.isObject(roomUser) && !_.isObject(room)) {
-					throw new Meteor.Error('invalid-channel');
-				}
-				if (!room) {
-					Meteor.runAsUser(user._id, function() {
-						Meteor.call('createDirectMessage', roomUser.username);
-						room = RocketChat.models.Rooms.findOneById(rid);
-					});
-				}
+				room = retrieveDirectMessageInfo({ userId: user._id, channel });
 				break;
 			default:
-				throw new Meteor.Error('invalid-channel-type');
+				//Try to find the room by id or name if they didn't include the prefix.
+				room = retrieveRoomInfo({ userId: user._id, channel: channelType + channel, ignoreEmpty: true });
+				if (room) {
+					break;
+				}
+
+				//We didn't get a room, let's try finding direct messages
+				room = retrieveDirectMessageInfo({ userId: user._id, channel: channelType + channel });
```

Comment:
since `retrieveDirectMessageInfo` can create a direct message room if it doesn't find a room by `_id` or `username` is this a expected behavior?